crates/storage-rocksdb/src/codec.rs

```diff
@@ -143,6 +144,27 @@ impl<M: Message + Default> Codec for ProtoValue<M> {
     }
 }
 
+impl Codec for InvocationUuid {
+    fn encode<B: BufMut>(&self, target: &mut B) {
+        let slice = self.as_bytes();
+        debug_assert_eq!(slice.len(), 16);
+        target.put_slice(slice);
+    }
+
+    fn decode<B: Buf>(source: &mut B) -> crate::Result<Self> {
+        // note: this is a zero-copy when the source is bytes::Bytes.
+        if source.remaining() < 16 {
+            return Err(StorageError::DataIntegrityError);
+        }
+        let bytes = source.copy_to_bytes(16);
```

Comment:
maybe replace `16`s with `self.serialized_length()`? it'll get inlined by the compiler anyway.

Reply:
Good catch. Will change it.